docs/use_cases/fraud_detection/fraud_detection.md

```diff
@@ -0,0 +1,516 @@
+
+The entire use case is implemented as a Python script written using the Feldera
+[Python SDK](https://www.feldera.com/python/).  It is available from our [github
+repository](https://github.com/feldera/feldera/blob/main/demo/project_demo10-FraudDetectionDeltaLake/run.py)
+and can be run from the command line or from your favorite Python notebook
+environment.
+
+## Credit card fraud detection
+
+Credit card fraud detection is a classic application of real-time feature
+engineering.  Here, data comes in a stream of transactions, each with attributes
+like card number, purchase time, vendor, and amount.  Additionally, the fraud
+detector has access to a slowly changing table with demographics information
+about cardholders, such as age and address.
+
+![Input tables and the output view](tables.png)
+
+## Input data
+
+We used a publicly available [Synthetic Credit Card Transaction
+Generator](https://github.com/namebrandon/Sparkov_Data_Generation) to generate
+two labeled datasets, both with 1000 user profiles.  We will use the first
+dataset for model training and testing, and the second dataset -- for real-time
+inference.  We stored the datasets in the [Delta Lake format](https://delta.io/)
+in two public S3 buckets:
+
+* Training dataset:
+  * Demographics table: `s3://feldera-fraud-detection-data/demographics_train/`
+  * Transaction table: `s3://feldera-fraud-detection-data/transaction_train/`
+
+* Inference dataset:
+  * Demographics table: `s3://feldera-fraud-detection-data/demographics_train/`
+  * Transaction table: `s3://feldera-fraud-detection-data/transaction_train/`
+
+## Model training and testing
+
+Finding an optimal set of features to train a good ML model is an iterative
+process.  At every step, the data scientist trains and tests a model using
+currently selected feature queries on an array of labeled historical data.  The
+results of each experiment drive the next refinement of feature queries.  In
+this scenario, feature vectors are computed in batch mode over a slice of
+historical data, e.g., data collected over a two-week timeframe.
+
+![Feature engineering as an iterative process](training_animation.gif)
+
+Below we walk the reader through one iteration of this process: we define a
+set of features, train a model using these features, and test its accuracy.
+
+### Writing feature queries
+
+We define several features over our input tables:
+
+* Data enrichment:
+  - We add demographic attributes, such as zip code, to each transaction
+* Rolling aggregates:
+  - average spending per transaction in the past day, week, and month
+  - average spending per transaction over a 3-month timeframe on the same day of the week
+  - number of transactions made with this credit card in the last 24 hours
+* Other:
+  - `is_weekend` - transaction took place on a weekend
+  - `is_night` - transaction took place before 6am
+  - `d` - day of week
+
+The following Python function uses the Feldera Python SDK to creates an SQL program,
```

Comment:
Create

Reply:
A

Reply:
From ChatGPT: "The correct usage is "an SQL." This is because the pronunciation of "SQL" starts with the vowel sound "ess," so it follows the rule of using "an" before words that begin with a vowel sound."